Sources/NIOPosix/NIOThreadPool.swift

```diff
@@ -292,6 +292,36 @@ extension NIOThreadPool {
     }
 }
 
+@available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+extension NIOThreadPool {
+    /// Runs the submitted closure if the thread pool is still active, otherwise throw an error.
+    /// The closure will be run on the thread pool so can do blocking work.
+    ///
+    /// - parameters:
+    ///     - body: The closure which performs some blocking work to be done on the thread pool.
+    /// - returns: result of the passed closure.
+    @preconcurrency
+    public func runIfActive<T>(_ body: @escaping @Sendable () throws -> T) async throws -> T {
+        try await self._runIfActive(body)
+    }
+    
+    private func _runIfActive<T>(_ body: @escaping () throws -> T) async throws -> T {
```

Comment:
I don't think we need this inner function now, and so we can avoid dropping the `@Sendable` annotation here.

Reply:
done